app/services/search/service.rb

```diff
@@ -0,0 +1,168 @@
+  #   results at once
+  # * Most drastically, pushing drafts and submissions to some unified search index
+
+  # (Although all of this may be obviated if we eventually decide to store drafts in the app store too)
+  class Service
+    def self.call(service, filters, query_params)
+      new(service, filters, query_params).call
+    end
+
+    attr_reader :service, :filters, :query_params
+
+    def initialize(service, filters, query_params)
+      @service = service
+      @filters = filters
+      @query_params = query_params
+    end
+
+    def call
+      local_unpaginated_results = query_local_drafts_without_pagination
+      app_store_response_with_paginated_results = query_app_store_with_pagination
+
+      if local_unpaginated_results.none?
+        [pagy(app_store_response_with_paginated_results.total_results), app_store_response_with_paginated_results.rows]
+      elsif app_store_response_with_paginated_results.none?
+        [pagy(local_unpaginated_results.count), paginate_local_results(local_unpaginated_results)]
+      else
+        combine_and_repaginate(local_unpaginated_results, app_store_response_with_paginated_results.total_results)
+      end
+    end
+
+    def pagy(total_results)
+      Pagy.new(
+        count: total_results,
+        limit: pagination_params[:per_page],
+        page: pagination_params[:page]
+      )
+    end
+
+    def query_local_drafts_without_pagination
+      Search::LocalService.call(base_query, filters)
+    end
+
+    def base_query
+      case service
+      when :nsm
+        Claim.for(filters[:current_provider]).includes(:defendants)
+      when :prior_authority
+        PriorAuthorityApplication.for(filters[:current_provider]).includes(:defendant)
+      else
+        raise "Unknown service #{service}"
+      end
+    end
+
+    def query_app_store_with_pagination
+      Search::AppStoreService.call(service, filters.merge(pagination_params))
+    end
+
+    def combine_and_repaginate(local_unpaginated_results, total_app_store_results)
+      app_store_unpaginated_results = Search::AppStoreService.call(service,
+                                                                   filters.merge(page: 1, per_page: total_app_store_results))
+      combined_results = app_store_unpaginated_results.rows + local_unpaginated_results.map { Search::Result.new(_1) }
+
+      [pagy(combined_results.count), order(combined_results).then { paginate(_1) }]
+    end
+
+    def paginate_local_results(local_unpaginated_results)
+      local_unpaginated_results.order(local_order)
+                               .offset(offset)
+                               .limit(pagination_params[:per_page])
+                               .map { Search::Result.new(_1) }
+    end
+
+    def order(combined_results)
+      sorted = combined_results.sort_by do |result|
+        raw = result.public_send(unified_sort_by)
+        raw.respond_to?(:downcase) ? raw.downcase : raw
```

Comment:
`raw.try(:downcase) || raw` or even `raw&.downcase || raw` maybe

Reply:
Good point, I had totally forgotten `try` existed. FWIW `&.` won't save me here, because it will blow up if `raw` exists but does not have a `downcase` method. But `try` is exactly the right tool for the job.